homeassistant/components/cast/media_player.py

```diff
@@ -218,6 +222,9 @@ async def async_added_to_hass(self):
         self._cast_view_remove_handler = async_dispatcher_connect(
             self.hass, SIGNAL_HASS_CAST_SHOW_VIEW, self._handle_signal_show_view
         )
+        self._cast_function_remove_handler = async_dispatcher_connect(
```

Comment:
I don't see anything using this?

Reply:
`_will_remove_from_hass`

Or is that correct usage?

Reply:
@eerovil I think the question is who will send `SIGNAL_HASS_CAST_APPLICATION`

Reply:
Oh, right. It's for extending the functionality. Cast extensions sometimes requires just launching a cast app by its ID. 

The signal isn't needed if the service would also support launching an arbitrary application (i.e. not supported by pychromecast, say netflix). Maybe I could add a possibility to input an "app_id" in place of "app_name"?

Reply:
Yes, that sounds like a simpler solution than adding the signal stuff.

However, is it useful to just start an app?
Don't you need a way to send messages to the app also?

Reply:
For apps using Android remote control for casting, the app is first via pychromecast for better UX. Otherwise there is no feedback for the action until ~15 seconds.

That's the only use case currently.

Reply:
Integrations should only offer external control via services or via functions in `__init__.py` from the integration. We can't keep track of random signal that might or might not be used by other integrations.

Reply:
So, this is now resolved, as I moved the functionality into the service?